pkg/bdb/hash_page.go

```diff
@@ -66,21 +67,22 @@ func HashPageValueContent(db *os.File, pageData []byte, hashPageIndex uint16, pa
 		if err != nil {
 			return nil, xerrors.Errorf("failed to parse page=%d: %w", currentPageNo, err)
 		}
-		if currentPage.PageType != OverflowPageType {
-			continue
```

Comment:
I think a simpler fix will be just to add 

> `currentPageNo = currentPage.NextPageNo`


here, instead of indenting a large block of text